enterprise_learner_portal/api/v1/serializers.py

```diff
@@ -81,3 +85,18 @@ def _get_course_overview(self, course_run_id):
                 return overview
 
         return None
+
+    def _get_course_run_url(self, request, course_run_id):
+        """
+        Get the appropriate course url while incorporating Executive Education content.
+        """
+        course_key = CourseKey.from_string(course_run_id)
+        course_key_str = '{}+{}'.format(course_key.org, course_key.course)
+        fields = ["course_type"]
+        course_data = get_course_data(course_key_str, fields)
+        if course_data:
+            course_type = course_data.get("course_type")
+            if course_type == EXEC_ED_COURSE_TYPE:
+                return request.build_absolute_uri(EXEC_ED_LANDING_PAGE)
```

Comment:
One way could be to fetch data for all the courses in a single api call and then parse and use it.

Reply:
We would need the course keys for all the enrollments ...